File: qimen/zhu.py
from dataclasses import dataclass
from qimen.jia_zi_graph import LiuShiJiaZi

@dataclass
class Zhu:
    gan: str
    zhi: str

    def __str__(self) -> str:
        return f"{self.gan}{self.zhi}"

    def get_gan(self) -> str:
        if self.gan == "甲":
            return LiuShiJiaZi[str(self)]["遁"]
        else:
            return self.gan

File: internal/model/zhu.go
package model

import "qimen/internal/data"

// 四柱
type Zhu struct {
	Gan string
	Zhi string
}

func (z Zhu) String() string {
	return z.Gan + z.Zhi
}

func (z Zhu) GetGan() string {
	for _, cycle := range data.LiuShiJiaZi {
		if z.String() == cycle.XunShou {
			return cycle.Dun
		}
	}
	return z.Gan
}
